addons/@fumohouse/fumo/fumo_appearances.gd

```diff
@@ -0,0 +1,92 @@
+class_name FumoAppearances
+extends Node
+## Singleton to keep track of Fumo appearance presets and to manage presets
+## for the local character.
+##
+## Keeps track of active and staging appearances, to use in-game and when
+## editing respectively.
+
+## Emitted after [member active] changes.
+signal active_changed(new_active: Appearance)
+## Emitted after [member staging] changes.
+signal staging_changed(new_staging: Appearance)
+## Emitted after [member presets] is updated.
+signal presets_updated
+
+## Current appearance for local character
+##
+## After modifying this value or its fields, it is recommended to call
+## [method active_emit] to notify listeners.
+var active: Appearance = preload(
+	"res://addons/@fumohouse/fumo_models/resources/presets/doremy.tres"
+)
+
+## Preview appearance to be applied.
+##
+## After modifying this value or its fields, it is recommended to call
+## [method staging_emit] to notify listeners.
+var staging: Appearance = active.duplicate(true)
+
+## List of presets available.
+var presets: Array[Appearance]
+
+
+## Get the singleton instance for this node.
+static func get_singleton() -> FumoAppearances:
+	return Modules.get_singleton(&"FumoAppearances") as FumoAppearances
+
+
+func _ready():
+	scan_dir("res://addons/@fumohouse/fumo_models/resources/presets")
+
+
+## Scan [param dir] recursively for model presets.
+## Remember to call [code]entries_updated.emit()[/code], after scanning
+## multiple directories
+func scan_dir(path: String):
+	var updated: bool = false
+
+	var dir := DirAccess.open(path)
+	if not dir:
+		push_error("Failed to open presets directory: '%'." % DirAccess.get_open_error())
+		return
```

Comment:
I only found this out recently, but this will not work in release mode because all resources are remapped to different paths. You need to use `ResourceLoader.list_directory`.